doc/user_quide.asciidoc

```diff
@@ -0,0 +1,105 @@
+# SUSE Distribution Migration System
+===========
+:Authors: Marcus Schäfer and Jesús Bermúdez Velázquez
+:Publication_Date: TBD
+:Latest_Version: 0.5.15
+:Contributors: 
+:Repo: https://github.com/SUSE/suse-migration-services[suse-migration-services]
+
+[%hardbreaks]
+Authors: {Authors}
+Contributors: {Contributors}
+Publication Date: {Publication_Date}
+Latest Version: {Latest_Version}
+Code available: {Repo}
+
+=== Concept
+Upgrading from one release to a newer one having to skip a range of multiple
+releases introduces an upgrade problem.
+
+The Distribution Migration System provides an upgrade path from A to B,
+where A and B are to major distributions versions (for example, from SLES 12 SP4 to SLES 15 GA).
+
+=== Requirements
+The system to be migrated must be registered.
+
+[listing]
+$ SUSEConnect --regcod [SLES_REGCODE]
+
+=== How does distribution migration system do it ?
+To achieve the migration between this two releases, the entire migration process
+is called from within a Live Migration Image.
```

Comment:
Which kernel / OS is used while in this Live Migration Image? (what to do in case the NIC or HDD needs a special or 3rd party driver?

Reply:
The live migration image is a build of the migration target. This means the SLE15-Migration package comes with a live migration system that is build against SLE15. This is also one of the requirements of the zypper migration plugin. You can migrate from e.g SLE12 to SLE15 only with the SLE15 version of zypper. So you can expect the live migration system to be always running in the target distro

> what to do in case the NIC or HDD needs a special or 3rd party driver

We can include whatever 3rd party driver is needed into the image build. But please note that the live migration image build is standard SLES and everything that we include must be available in some repo and SLE compliant

Reply:
> We can include whatever 3rd party driver is needed into the image build. But please note that the live migration image build is standard SLES and everything that we include must be available in some repo and SLE compliant

Do we have a documentation on how to rebuild and customize the ISO so that a customer or partner could add third party drivers on his own?
I do not believe that we really have this problem today - as the servers customers are using are SLES 12 SP* and YES certified hardware should be used - so expectation is that the same hardware works for SLES 15 SP* - but you never know for sure ;-).

Reply:
There is no docs on how to rebuild the live iso. The live iso build is packaged into an rpm. So the customer has access to the rpm and binary blob inside. The building of that image is in the internal buildservice in a project. If you are a developer at suse you can simply run ```osc branch _project_``` and just add the needed modifications. In terms of 3rd party stuff that would mean adding an enterprise buildservice repo and some packages. The problem I see here is that we do not populate that build to the outside. Imho this is a complete new topic in which we could discuss that we also package the live image build descriptions and put some documentation into that package such that a customer can build them on their own. To be honest I don't want customers to build that live image to drive an upgrade process. The reason is that you can do all kinds of nasty things in that image and if we open this gate we also should tell customers that we don't take over any warranty. If we do that it means we don't care for the upgrade process and that in the end will not be helpful to customers. I see this part more as a service. A customer with special needs gets in contact with the suse consulting and then we build based on what we have. I see this more as a business opportunity

Reply:
Yeah - agreed. Might be Pandoras box.. lets go with L3 for the moment and see what comes in over time.

Reply:
Customers that have a physical machine should continue to use the "boot from install ISO" method. I don't see us covering the general case of "use this anywhere SLES may run and with whatever configuration you might have". The install ISO handles this we have no intention of duplicating that functionality.

Reply:
> 
> 
> Customers that have a physical machine should continue to use the "boot from install ISO" method. I don't see us covering the general case of "use this anywhere SLES may run and with whatever configuration you might have". The install ISO handles this we have no intention of duplicating that functionality.

At least for the SES usecase we have to support this method also for hardware based installations (SES is 99% on hardware). 
The "alternative method for automation of the upgrade using autoyast based upgrade or even manual upgrades using the DVD would be far to complicated and can not be standardized among all the different customer deployments out there which have different and complicated network setups".
However - lets do one step after the other and see where it goes - in the worst case we can still fallback to autoyast based or manual DVD based upgrades if this method does not work.